mjattbot.js

```diff
@@ -58,33 +60,42 @@ function pokeCatch() {
 }
 
 client.on("message", message => {
-  if (message.author.bot) return;
-  if (message.channel.type === "dm") {
-    message.channel.send(
-      "Please try this in a public guild. Error: DM channel"
-    );
-    return;
-  }
-  if (!message.content.startsWith(key.prefix)) return;
+  if (message.author.bot) return; // ignore bots to avoid infinite loop
+  if (message.channel.type === "dm") return message.channel.send("Please try this in a public guild. Error: DM channel"); // cleanup, code looks bit cleaner here imo
+  
+  if (!message.content.startsWith(key.prefix)) return; // ignore msgs without prefix
 
   let command = message.content.split(" ")[0];
   command = command.slice(key.prefix.length);
+  // console.log(command);
 
   let args = message.content.split(" ").slice(1);
 
   try {
     let cmd = require(`./commands/${command}.js`);
-    cmd.run(client, message, args);
     count++;
     console.log(
-      `Message Number: ${count} Command: ${command} Args: ${args.join(" ")} Author: ${message.author.username}`
+      `Message Number: ${count} Command: ${command} Args: ${args.join(" ") ? args.join(" ") : "no arguments"} Author: ${message.author.username}`
     );
+    cmd.run(client, message, args);
   } catch (err) {
-    console.log(err);
+    try {
+      let tagcmd = require(`./commands/tag.js`); // require the tag command
+      let args = message.content.replace(key.prefix, "").split(" "); // makes sure that in this case, args for the tag command is the message content without the prefix
+      // console.log(args);
+      tagcmd.run(client, message, args);
+    } catch (err) {
+      console.error(err);
+    }
   }
 });
-client.login(key.token); //the token used to initiate the bot, KEEP SECRET PLS
+client.login(key.token).then(() => {
+  console.info(`Logged in successfully as ${client.user.tag}`);
+}).catch(err => console.error("Failed to login: " + err)); /* the token used to initiate the bot, KEEP SECRET PLS 
+also added catching error with failing login
+*/
 
 process.on("unhandledRejection", err => {
-  console.error("Unhandled exception: " + err.stack);
+  console.error("Unhandled promise rejection, make sure to catch promise rejections!\n " + err.stack); /* added a reminder
```

Comment:
I don't really see what this is adding, its a ton more characters for no real gain.

Reply:
It's more just a reminder to catch promise errors. Can also switch that to just `console.error(err.stack)`